### PR TITLE
Update SHOW GRANTS: object_name, is_grantable

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-10126, DOC-3654, DOC-3655

NB. These changes are applied to v23.1, v23.2, v24.1, v24.2

### CODE REVIEW

**Comment by rmloveland** [comment]
tl;dr: @Dedej-Bergin this PR is kinda big but the relevant part for you is probably just the subset of the diff on `v24.2/show-grants.md` that changes `SHOW GRANTS` outputs to use `object_name`, etc.  All of the changes to each version are identical and much of the other content is "necessary updates to grants/roles/etc docs that were not caused by cockroachdb/cockroach#122197" so you can ignore all the other stuff IMO

commentary below is added context, probably more for the eventual docs reviewer @taroface than for @Dedej-Bergin:

The majority of the other changes in this PR are docs housekeeping re: updating the output of various role/grant/etc. commands which were badly in need of updating.  A lot of it is silliness about `true` -> `t`, but some is more serious, like how there are way more role options now, so the old examples didn't work anymore because they assumed if you add 2 role options for user X on table Y and then drop those 2 role options on user X, now the table Y doesn't have any more stuff granted to user X. Unfortunately this is not true anymore so that code stopped working.  So I simplified the examples to remove those assumptions.  I also did a "refactoring" change w.r.t deleting redundant tutorial content from `authorization.md` which is already being maintained in the various `grant.md`, etc. files

I'm sorry for the big PR, a lot of the SQL wasn't working at all anymore and badly needed updates so the scope grew as I went along, feel free to yell at me about this

**Review by Dedej-Bergin — APPROVED** [review]
Thanks @rmloveland , the show-grants.md changes look good!

File: src/current/v23.1/grant.md [hunk 1]
```diff
@@ -138,32 +139,24 @@ SHOW GRANTS ON TABLE movr.public.*;
...
   movr          | public      | vehicles                   | admin   | ALL            |      t
-  movr          | public      | vehicles                   | demo    | ALL            |      t
   movr          | public      | vehicles                   | max     | ALL            |      f
-  movr          | public      | vehicles                   | public  | SELECT         |      f
   movr          | public      | vehicles                   | root    | ALL            |      t
-(26 rows)
+(18 rows)
 ~~~
 
 To ensure that anytime a new table is created, all the privileges on that table are granted to a user, use [`ALTER DEFAULT PRIVILEGES`]({% link {{ page.version.version }}/alter-default-privileges.md %}):
```
Comment (taroface) [inline src/current/v23.1/grant.md:162]: I cannot let this one pass

```suggestion
To ensure that any time a new table is created, all the privileges on that table are granted to a user, use [`ALTER DEFAULT PRIVILEGES`]({% link {{ page.version.version }}/alter-default-privileges.md %}):
```

**Review by taroface — COMMENTED** [review]
Just a few comments and nits. I commented on 23.1 but they should apply across versions.

File: src/current/v23.1/revoke.md [hunk 2]
```diff
@@ -303,32 +293,12 @@ SHOW GRANTS ON TYPE status;
...
-  movr          | public      | status    | admin   | ALL
-  movr          | public      | status    | max     | ALL
-  movr          | public      | status    | public  | USAGE
-  movr          | public      | status    | root    | ALL
-(4 rows)
-~~~
-
-{% include_cached copy-clipboard.html %}
-~~~ sql
-REVOKE GRANT ON TYPE status FROM max;
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/v23.1/revoke.md:317]: you're right, i added it back in, this was an oversight on my part. thanks for catching it!

File: src/current/v23.1/authorization.md [hunk 3]
```diff
@@ -27,447 +28,23 @@ A new user must be granted the required privileges for each database and table t
 By default, a new user belongs to the `public` role and has no privileges other than those assigned to the `public` role.
 {{site.data.alerts.end}}
 
-
 ## Create and manage roles
 
-To create and manage your cluster's roles, use the following statements:
+For examples showing how to create and manage your cluster's users and roles, see the following documentation:
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/v23.1/authorization.md]: updated! i am still "seeing" things "above" and "below"! will keep working on that

File: src/current/v23.1/drop-user.md [hunk 4]
```diff
@@ -38,51 +38,74 @@ In this example, first check a user's privileges. Then, revoke the user's privil
...
+  database_name | schema_name | table_name | grantee | privilege_type | is_grantable
+----------------+-------------+------------+---------+----------------+---------------
+  test          | public      | customers  | max     | ALL            |      f
+(1 row)
 ~~~
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> DROP USER mroach;
+REVOKE CREATE,INSERT,UPDATE ON customers FROM max;
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply src/current/v23.1/drop-user.md:61]: i revised this a bit so hopefully it all works together now. the reason this is awkward is b/c the original docs explicitly removed those 3 privs when they were the only 3, so the commands in this section no longer worked.

along the way I decided to try to show that unless you drop ALL the privs you can't drop the user yet

def open to more feedback.  if you think it's too wordy, we could also do the simple example of `GRANT ALL` / `REVOKE ALL` and call it good

PTAL!

File: src/current/v23.1/grant.md (hunk 1 above)
Comment (rmloveland) [reply src/current/v23.1/grant.md]: the language is evolving! the words are growing together! it's like 'email' (https://english.stackexchange.com/questions/1925/email-or-e-mail)

jkjk fixed it (for now ....)

**Comment by rmloveland** [comment]
@taroface i took a shot at updating with your feedback in the latest commit. to keep it small i only updated v23.1 so we can hash that out. once it's good i will port it to all versions

File: src/current/v23.1/drop-user.md [hunk 5]
```diff
@@ -30,59 +30,80 @@ See [`DROP ROLE`: Synopsis]({% link {{ page.version.version }}/drop-role.md %}#s
...
-| customers | mroach | UPDATE     |
-+-----------+--------+------------+
-(3 rows)
+  database_name | schema_name | table_name | grantee | privilege_type | is_grantable
+----------------+-------------+------------+---------+----------------+---------------
+  test          | public      | customers  | max     | ALL            |      f
+(1 row)
 ~~~
 
+We can test that dropping the user will fail unless all privileges are revoked (including [default privileges]({% link {{ page.version.version }}/security-reference/authorization.md %}#default-privileges)). Issue the following statement to revoke a subset of the user's privileges:
```
Comment (taroface) [inline src/current/v23.1/drop-user.md:57]:
```suggestion
You can test that dropping the user will fail unless all privileges are revoked (including [default privileges]({% link {{ page.version.version }}/security-reference/authorization.md %}#default-privileges)). Issue the following statement to revoke a subset of the user's privileges:
```

File: src/current/v23.1/drop-user.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (taroface) [reply src/current/v23.1/drop-user.md:61]: Looks good!

**Review by taroface — APPROVED** [review]
Looks good! Just a nit ("you" versus "we").

File: src/current/v23.1/drop-user.md (hunk 5 above)
Comment (rmloveland) [reply src/current/v23.1/drop-user.md:57]: Fixed!
